src/main/scala/Util.scala

```diff
@@ -78,3 +78,33 @@ class PutSeqDriver(val s: Seq[Tuple2[BigInt,Int]])(implicit p: Parameters) exten
 
 }
 
+// Use gray coding to safely synchronize a word across a clock crossing.
+// This should be placed in the receiver's clock domain.
+class WordSync[T <: Data](gen: T, lat: Int = 2) extends Module {
+  val size = gen.cloneType.fromBits(UInt(0)).asUInt().getWidth
+  val io = new Bundle {
+    val in = gen.cloneType.flip
```

Comment:
Yeah I got the right version of chisel just by merge from the hwacha branch of rocket-chip (but not before)